### PR TITLE
remove org.uncommons.maths dependency (#12636)

### PR DESCRIPTION
**Pull request by gosubpl**
and also add `@InternalApi` annotation

### CODE REVIEW

**Comment by gosubpl** [comment]
Refs #12636

**Comment by gosubpl** [comment]
WIP - needs work on `akka-samples-osgi-dinning-hakkers` as it contains `org.uncommons.maths` as dependency too

**Comment by patriknw** [comment]
@gosubpl All samples have been moved to https://github.com/akka/akka-samples and will be removed from akka/akka
Please do the changes there, but if it's a stopper you could update akka/akka also.

**Comment by gosubpl** [comment]
This is followup PR to https://github.com/akka/akka/pull/22013

**Comment by gosubpl** [comment]
@patriknw thanks for the info. So I will leave the samples in `akka/akka` as they are, since they are going to be removed anyway. That is not a stopper, since this is done via separate `pom.xml`. As to the `akka/akka-samples` this is not possible now, as they depend on `<akka.version>2.4.14</akka.version>` and `2.4` still uses `org.uncommon.maths`. As soon as the samples move to `2.5` I will submit a PR to remove it from there.
I still need to add `setDaemon(true)` to the re-seeder thread, though, as per your comment in #22013 .

**Comment by gosubpl** [comment]
added `setDaemon(true)`
no longer WIP.

File: akka-remote/src/main/scala/akka/remote/security/provider/AESCounterBuiltinRNG.scala [hunk 1]
```diff
@@ -127,5 +132,9 @@ private object CounterRNGConstants {
 }
 
 private class AESCounterBuiltinRNGReSeeder extends ThreadFactory {
-  override def newThread(r: Runnable): Thread = new Thread(r, "AESCounterBuiltinRNGReSeeder")
+  override def newThread(r: Runnable): Thread = {
+    val thread = new Thread(r, "AESCounterBuiltinRNGReSeeder")
+    thread.setDaemon(true)
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/security/provider/AESCounterBuiltinRNG.scala]: +1

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by ktoso** [comment]
Ok, with that last fix we're good to go here I believe :-)

**Comment by ktoso** [comment]
LGTM :-)

**Pull request merged**